tools/plugins/mgm/src/main/kotlin/net/corda/cli/plugins/mgm/SetupNetwork.kt

```diff
@@ -0,0 +1,167 @@
+        kubectl("create", "ns", networkName)
+        kubectl("label", "ns", networkName, "namespace-type=corda-e2e", "--overwrite=true")
+
+        try {
+            kubectl(
+                "create", "secret", "docker-registry", "docker-registry-cred",
+                "--docker-server=corda-os-docker.software.r3.com",
+                "--docker-username=${System.getenv("CORDA_ARTIFACTORY_USERNAME")}",
+                "--docker-password=${System.getenv("CORDA_ARTIFACTORY_PASSWORD")}",
+                "-n", networkName
+            )
+        } catch (e: SetupException) {
+            // Do nothing, the secret might already be there
+        }
+
+        helm(
+            "install", "prereqs", "oci://corda-os-docker.software.r3.com/helm-charts/corda-dev",
+            "--set", "kafka.replicaCount=$kafkaReplicas,kafka.zookeeper.replicaCount=$zooKeeperReplicas",
+            "-n", networkName, "--wait", "--timeout", "600s"
+        )
+
+        val chart = File(cordaOsRuntimeDir, "charts")
+        val cordaChart = File(chart, "corda")
+        val yaml = File(File(File(cordaOsRuntimeDir, ".ci"), "e2eTests"), "corda.yaml")
+        helm(
+            "install",
+            "corda", cordaChart.absolutePath,
+            "-f", yaml.absolutePath,
+            "--set", "image.tag=$actualBaseImage,bootstrap.kafka.replicas=$kafkaReplicas,kafka.sasl.enabled=false",
+            "-n", networkName, "--wait", "--timeout", "600s"
+        )
+    }
+
+    private val actualBaseImage by lazy {
+        if (baseImage == null) {
+            val tagsReply = Unirest.get(
+                "https://software.r3.com:443/v2/corda-os-docker-unstable/corda-os-p2p-link-manager-worker/tags/list"
+            )
+                .basicAuth(
+                    System.getenv("CORDA_ARTIFACTORY_USERNAME"),
+                    System.getenv("CORDA_ARTIFACTORY_PASSWORD"),
+                ).asJson()
+            if (!tagsReply.isSuccess) {
+                throw SetupException("Could not get list of tags. ${tagsReply.statusText}")
+            }
+            val tagsList = tagsReply.body.`object`?.get("tags")
+                as? JSONArray
+                ?: throw SetupException("Tags has the wrong format - ${tagsReply.body}")
+            tagsList.filterNotNull()
+                .map {
+                    it.toString()
+                }.filter { it.startsWith(PREFIX) }.maxByOrNull {
+                    it.removePrefix(PREFIX).toLongOrNull() ?: 0
+                }?.also {
+                    println("Using base image: $it")
+                } ?: throw SetupException("Could not find any tag")
+        } else {
+            baseImage
+        }
+    }
+
+    override fun run() {
+        if (minikube) {
+            kubectl("config", "use-context", "minikube")
+        }
+        if (awsName != null) {
+            execute(
+                "aws",
+                "eks",
+                "update-kubeconfig",
+                "--name",
+                "eks-e2e"
```

Comment:
Should this be using the input?
```suggestion
                awsName
```